features/data_converter/encryption/codecs/data_converter.go

```diff
@@ -0,0 +1,150 @@
+package codecs
+
+// Adapted from samples-go/encryption/data_converter.go and crypto.go
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"fmt"
+	"io"
+
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/sdk/converter"
+)
+
+const (
+	MetadataEncodingEncrypted = "binary/encrypted"
+	MetadataEncryptionKeyID   = "encryption-key-id"
+	TEST_KEY_STR              = "test-key-test-key-test-key-test!"
+)
+
+type DataConverter struct {
+	converter.DataConverter
+	options DataConverterOptions
+}
+
+type DataConverterOptions struct {
+	KeyID string
+}
+
+// NewEncryptionDataConverter creates a new instance of EncryptionDataConverter wrapping a DataConverter
+func NewEncryptionDataConverter(dataConverter converter.DataConverter, options DataConverterOptions) *DataConverter {
+	codecs := []converter.PayloadCodec{
+		&Codec{KeyID: options.KeyID},
+	}
+
+	return &DataConverter{
+		DataConverter: converter.NewCodecDataConverter(dataConverter, codecs...),
+		options:       options,
+	}
+}
+
+// Codec implements PayloadCodec using AES Crypt.
+type Codec struct {
+	KeyID string
+}
+
+func (e *Codec) getKey(keyID string) (key []byte) {
```

Comment:
Not sure why you need this method.
In a real payload the signature for getting a key would return an error.
Also note the unused param here, maybe return []byte(keyID) (plus some prefix / suffix).